tensorzero-core/src/tool.rs

```diff
@@ -325,6 +326,12 @@ impl std::fmt::Display for ToolCall {
     }
 }
 
+impl RateLimitedInputContent for ToolCall {
+    fn estimated_input_token_usage(&self) -> u64 {
+        get_estimated_tokens(&self.name) + get_estimated_tokens(&self.arguments)
```

Comment:
Can you make this (and all of the other impls of `estimated_input_token_usage`) do an explicit `let ToolCall { name, arguments } = self`? This will cause the compiler to error if we forget to mention any new fields in these methods.

Reply:
done everywhere